cfme/tests/middleware/test_middleware_deployment.py

```diff
@@ -132,3 +149,228 @@ def test_tags(provider):
+def test_start(provider, archive):
+    """Tests Starting of stopped archive into EAP7 server
+
+    Steps:
+        * Get servers list from UI
+        * Chooses JBoss EAP server from list
+        * Deploys some deployment archive into server.
+        * Stops that archive.
+        * Refreshes the provider.
+        * Lists all deployments on EAP server.
+        * Verified the recently stopped archive's status is Enabled.
+        * Selects that archive from the list to load details.
+        * Verifies that properties of deplyment's summary page and the status is Enabled.
+    """
+    server = get_eap_server(provider)
+    runtime_name = deploy(provider, server, archive)
+    check_deployment_enabled(provider, server, runtime_name)
+    deployment = get_deployment_from_list(provider, server, runtime_name)
+    deployment.stop()
+    check_deployment_disabled(provider, server, runtime_name)
+    deployment.start()
+    # enable when #10111 is fixed
+    # _check_deployment_enabled(provider, server, runtime_name)
+    # deployment = _get_deployment_from_list(provider, server, runtime_name)
+    # deployment.validate_properties()
+
+
+def get_deployments_set(deployments):
+    """
+    Return the set of deployments which contains only necessary fields,
+    such as 'name' and 'server'
+    """
+    return set((deployment.name, deployment.server.name) for deployment in deployments)
+
+
+def get_deployments_statuses(deployments):
+    """
+    Return the map of deployments which contains,
+    'name' as key, 'status' as value
+    """
+    return {deployment.name: deployment.status for deployment in deployments}
+
+
+def deploy(provider, server, file_path):
+    runtime_name = "{}_{}".format(fauxfactory.gen_alpha(8).lower(), os.path.basename(file_path))
+    server.add_deployment(file_path, runtime_name)
+    provider.refresh_provider_relationships(method='ui')
+    return runtime_name
+
+
+def get_eap_server(provider):
+    server = None
+    server_list = MiddlewareServer.servers(provider=provider)
+    for _server in server_list:
+        if _server.product == 'JBoss EAP':
+            server = _server
+            break
+    assert server, 'EAP server was not found in servers list'
+    return server
+
+
+def get_local_server(provider):
+    server = None
+    server_list = MiddlewareServer.servers(provider=provider)
+    for _server in server_list:
+        if _server.product == 'Hawkular':
+            server = _server
+            break
+    assert server, 'Hawkular server was not found in servers list'
+    return server
+
+
+def get_deployment_from_list(provider, server, runtime_name):
+    deployment = None
+    for _deployment in MiddlewareDeployment.deployments(provider=provider, server=server):
+        if _deployment.name == runtime_name:
+            deployment = _deployment
+            break
+    assert deployment, 'Recently deployed archive was not found in deployments list'
+    return deployment
```

Comment:
You can use the comment above to refactor this function in similar manner. The logic is exactly the same.